packages/samples/headless-react/src/components/quickview/quickview.fn.tsx

```diff
@@ -0,0 +1,49 @@
+import {useEffect, useState, useContext, FunctionComponent} from 'react';
+import {buildQuickview, Result} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+interface QuickviewProps {
+  result: Result;
+}
+
+export const Quickview: FunctionComponent<QuickviewProps> = (props) => {
+  const {result} = props;
+  const {engine} = useContext(AppContext);
+
+  const controller = buildQuickview(engine!, {options: {result}});
```

Comment:
FYI @btaillon, I was able to create the quickview controller inside the function component today. This was not the case when I worked on the search-box sample, where React would go into an infinite loop.

Not sure what changed, but we have an opportunity to make our functional samples even nicer. We could divide the work to migrate them over,

Reply:
I believe this is a side benefit of putting the engine in the context.

I like the idea of doing this migration, let's do it.